lib/Sema/TypeChecker.cpp

```diff
@@ -691,6 +692,7 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
       TC.processREPLTopLevel(SF, TLC, StartElem);
 
     typeCheckFunctionsAndExternalDecls(TC);
+    performTopLevelDeclDiagnostics(SF);
```

Comment:
…and now that I look at it a second time, does it make sense to sink it into typeCheckTopLevelCodeDecl? That way we don't have to walk things twice.